# Case Study on Testing of Hypothesis

A company started to invest in digital marketing as a new way of their product
promotions. For that they collected data and decided to carry out a study on it.

In [24]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

Reading the dataset

In [25]:
df =pd.read_csv('Sales_add.csv')

In [26]:
df

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [27]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [28]:
df.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

# The company wishes to clarify whether there is any increase in sales after stepping into digital marketing

We have to consider data before and after stepping into digital marketing

In [31]:
dfbef = df['Sales_before_digital_add(in $)']
dfaft = df['Sales_After_digital_add(in $)']

In [35]:
dfbef.describe()

count        22.000000
mean     149239.954545
std       14844.042921
min      130263.000000
25%      138087.750000
50%      147444.000000
75%      157627.500000
max      178939.000000
Name: Sales_before_digital_add(in $), dtype: float64

In [36]:
dfaft.describe()

count        22.000000
mean     231123.727273
std       25556.777061
min      187305.000000
25%      214960.750000
50%      229986.500000
75%      250909.000000
max      276279.000000
Name: Sales_After_digital_add(in $), dtype: float64

We have to check for any increase in sales. Here we have to consider a one-tailed t-test

NULL HYPOTHESIS: Sales before stepping into digital marketing is equal to Sales after stepping into digital market.

ALTERNATIVE HYPOTHESIS: Sales after stepping into digital marketing is greater than Sales before stepping into digital marketing

Significance level is assumed to be 0.05

In [34]:
stats.ttest_ind(a=dfbef, b=dfaft, equal_var=True)

Ttest_indResult(statistic=-12.995084451110877, pvalue=2.614368006904645e-16)

p value less than 0.05 means null hypothesis should be rejected. So we can conclude that sales increases after stepping into digital marketing

# The company needs to check whether there is any dependency between the features “Region” and “Manager”.

Here we have to check dependancy of two categorical variables. So we can do Chi square test here.

NULL HYPOTHESIS: Features Region and Manager are independant.

ALTERNATIVE HYPOTHESIS: Features Region and Manager are dependent. 

Significance level is assumed to be 0.05

In [30]:
df1 = df['Region']
df2 = df['Manager']

In [7]:
crosstab = pd.crosstab(df1, df2)
crosstab

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [8]:
from scipy.stats import chi2_contingency

In [9]:
chiVal, pVal, df, exp = chi2_contingency(crosstab)
chiVal, pVal, df, exp

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

We get p value as 0.549399 which is greater than 0.05

Greater p value means null hypothesis cannot be rejected. That is Region and Manager are independant features.

# Thank You!